ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -189,7 +193,12 @@ public void start() {
         bootstrapNodes.stream()
             .filter(this::whitelistIfPresentIsNodePermitted)
             .collect(Collectors.toList());
-    recursivePeerRefreshState.start(initialDiscoveryPeers, localPeer.getId());
+    if (nodePermissioningController.delayPeerDiscovery()) {
+      nodePermissioningController.peerDiscoveryCallback(
```

Comment:
When we dealy peer discovery, we should set the callback that will be triggered after we reach the sync status. In this case, the callback will be enabling peer discovery through `recursivePeerRefreshState.start`.